Updated release notes for NUnit 4.3

OsirisTerje opened the pull request.

SeanKilleen commented:
LGTM so far, but not merging because the title still has WIP so I assume @OsirisTerje has it still in progress.

OsirisTerje commented:
@SeanKilleen @manfred-brands @stevenaw @mikkelbu 

I have changed the release notes to be different with respect to acknowledgements.  Does it look ok to you?

I have picked a bit from how @mikkelbu has been doing it, but keeping the individuals for each PR, but also added the individual team members who contributed.  (In Norway, we often overlook the individual when celebrating the achievements of the team :-)  Trying to improve on that. )

mikkelbu commented on docs/articles/nunit/release-notes/framework.md:
```diff
@@ -2,9 +2,151 @@
+This release includes a series of enhancements and bug fixes. Notably, some of the bug fixes address issues related to a lack of type safety in NUnit. See issues [4877](https://github.com/nunit/nunit/issues/4877),
+[4876](https://github.com/nunit/nunit/issues/4876), [4875](https://github.com/nunit/nunit/issues/4875), [4874](https://github.com/nunit/nunit/issues/4874) and [150](https://github.com/nunit/nunit/issues/150).  
+
+Be aware that this might cause compile errors in cases where the code was previously ignored. These errors can be safely resolved by removing the affected code..
```
```suggestion
Be aware that this might cause compile errors in cases where the code was previously ignored. These errors can be safely resolved by removing the affected code.
```

mikkelbu commented on docs/articles/nunit/release-notes/framework.md:
```diff
@@ -2,9 +2,151 @@
+* [4836](https://github.com/nunit/nunit/issues/4836) PropertiesComparer doesn't work well with records. Thanks to [Oleksandr Liakhevych](https://github.com/Dreamescaper) for [PR 4837](https://github.com/nunit/nunit/pull/4837)
+* [4807](https://github.com/nunit/nunit/issues/4807) Is.EqualTo with empty ValueTuple throws System.NotSupportedException "Specified Tolerance not supported for instances of type 'System.ValueTuple' and 'System.ValueTuple'" after updating to NUnit 4.2.2. Thanks to NUnit Team member [Manfred Brands](https://github.com/manfred-brands) for [PR 4809]( https://github.com/nunit/nunit/pull/4809)
+* [4281](https://github.com/nunit/nunit/issues/4281) Throws and Delayed (.After) Constraints do not cooperate, resulting in incorrectly failing test. Thanks to [Oleksandr Liakhevych](https://github.com/Dreamescaper) for [PR 4815](https://github.com/nunit/nunit/pull/4815)
+* [3772](https://github.com/nunit/nunit/issues/3772) IsEquivalent on default ImmutableArray throws exception.. Thanks to [Sergio L.](https://github.com/mr-sergito) for [PR 4850](https://github.com/nunit/nunit/pull/4850)
```
I'm guessing the `..` is due to the title of the ticket ending with `.` ?

OsirisTerje replied:
Yes, thanks for noticing!

OsirisTerje replied:
Fixed

mikkelbu commented on docs/articles/nunit/release-notes/framework.md:
```diff
@@ -2,9 +2,151 @@
+### Enhancements
+
+* [4890](https://github.com/nunit/nunit/issues/4890) Add .net 8 as a target framework in the package. Thanks to NUnit Team member [Manfred Brands](https://github.com/manfred-brands) for [PR 4884]( https://github.com/nunit/nunit/pull/4884)
+* [4886](https://github.com/nunit/nunit/issues/4886) Optimize ToArray()[0] with First() in TestAssert class in TestAssert.cs. Thanks to [](https://github.com/Saurav-shres) for [PR 4888](https://github.com/nunit/nunit/pull/4888)
```
`Thanks to []` will not show anything in the resulting markdown

OsirisTerje replied:
Hmm, something amiss there, if no name it should put the account id there instead.

OsirisTerje replied:
Fixed

mikkelbu commented on docs/articles/nunit/release-notes/framework.md:
```diff
@@ -2,9 +2,151 @@
+* [4839](https://github.com/nunit/nunit/issues/4839) Allow to use collection expressions with collection constraints. Thanks to [Oleksandr Liakhevych](https://github.com/Dreamescaper) for [PR 4840](https://github.com/nunit/nunit/pull/4840)
+* [4832](https://github.com/nunit/nunit/issues/4832) Setting thread names to make debugging easier. Thanks to NUnit Team member [Terje Sandstrom](https://github.com/OsirisTerje) for [PR 4835]( https://github.com/nunit/nunit/pull/4835)
+* [4811](https://github.com/nunit/nunit/issues/4811) Assert.ThatAsync does not support polling. Thanks to [Oleksandr Liakhevych](https://github.com/Dreamescaper) for [PR 4813](https://github.com/nunit/nunit/pull/4813)
+* [4771](https://github.com/nunit/nunit/issues/4771) Support tuples in inline data. Thanks to [](https://github.com/MaxKot) for [PR 4772](https://github.com/nunit/nunit/pull/4772)
```
`Thanks to []` will not show anything in the resulting markdown

OsirisTerje replied:
Fixed

mikkelbu commented on docs/articles/nunit/release-notes/framework.md:
```diff
@@ -2,9 +2,151 @@
+* [4807](https://github.com/nunit/nunit/issues/4807) Is.EqualTo with empty ValueTuple throws System.NotSupportedException "Specified Tolerance not supported for instances of type 'System.ValueTuple' and 'System.ValueTuple'" after updating to NUnit 4.2.2. Thanks to NUnit Team member [Manfred Brands](https://github.com/manfred-brands) for [PR 4809]( https://github.com/nunit/nunit/pull/4809)
+* [4281](https://github.com/nunit/nunit/issues/4281) Throws and Delayed (.After) Constraints do not cooperate, resulting in incorrectly failing test. Thanks to [Oleksandr Liakhevych](https://github.com/Dreamescaper) for [PR 4815](https://github.com/nunit/nunit/pull/4815)
+* [3772](https://github.com/nunit/nunit/issues/3772) IsEquivalent on default ImmutableArray throws exception.. Thanks to [Sergio L.](https://github.com/mr-sergito) for [PR 4850](https://github.com/nunit/nunit/pull/4850)
+* [3713](https://github.com/nunit/nunit/issues/3713) Description of --where=EXPRESSION missing from Notes section. Thanks to [](https://github.com/rsjackson3) for [PR 4817](https://github.com/nunit/nunit/pull/4817)
```
`Thanks to []` will not show anything in the resulting markdown

OsirisTerje replied:
Fixed

mikkelbu reviewed (approved):
I like it. I only have some nit-pick, questions, and 3 instances of markdown generating `Thanks to for PR...` (so no mention of the contributor)

manfred-brands reviewed (approved):
I like it to. Nothing to add to @mikkelbu review comments,
